### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Github now allows to run CodeQL during pull request verification. This allows to detect errors / security problems early.

Modification:

Add config

Result:

Analysis during build

### CODE REVIEW

**Comment by artem-smotrakov** [comment]
Looks like the Ubuntu container, where the job runs, doesn't have some required deps

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # - name: Autobuild
+    #  uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
```
Comment (artem-smotrakov) [inline .github/workflows/codeql-analysis.yml:56]:
```suggestion
    - name: Install deps
      run: sudo apt-get install libapr1
```

Not sure if it needs only `libapr1`.

Comment (normanmaurer) [reply .github/workflows/codeql-analysis.yml:56]: I hope I added all of these... lets see :D

**Comment by normanmaurer** [comment]
@artem-smotrakov seems like this did the trick :)

**Comment by artem-smotrakov** [comment]
@normanmaurer Awesome! Now we have CodeQL checks here as well :+1:

**Review by artem-smotrakov — APPROVED** [review]

**Pull request merged**